miio/gateway.py

```diff
@@ -126,58 +125,64 @@ def light(self) -> "GatewayLight":
         """Return light control interface."""
         return self._light
 
-    @command()
+    @property
     def devices(self):
-        """Return list of devices."""
+        """Return a list of the already discovered devices."""
+        return self._devices
+
+    @command()
+    def discover_devices(self):
+        """Discovers SubDevices and returns a list of the discovered devices."""
         # from https://github.com/aholstenson/miio/issues/26
-        devices_raw = self.send("get_device_prop", ["lumi.0", "device_list"])
-        devices = [
-            SubDevice(self, *devices_raw[x : x + 5])  # noqa: E203
-            for x in range(0, len(devices_raw), 5)
-        ]
+        device_type_mapping = {
+            "AqaraRelayTwoChannels": AqaraRelayTwoChannels,
+            "Plug": AqaraPlug,
+            "SensorHT": SensorHT,
+            "AqaraHT": AqaraHT,
+            "AqaraMagnet": AqaraMagnet,
+        }
+        devices_raw = self.get_gateway_prop("device_list")
+        self._devices = []
+
+        for x in range(0, len(devices_raw), 5):
+            try:
+                device_name = DeviceType(devices_raw[x + 1]).name
```

Comment:
in principle we could just map the number directly to the class, but that gets confusing realy fast since you would constantly need to look up which number represents which device...

What do you prefer, schould I change it to a direct number mapping and still have the enum only for debug messages etc?

Reply:
Hmm, I don't understand what do you mean by checking the number? Wouldn't the type information be enough?

Reply:
ah right, I misunderstood your first message, I think I now know what you mean
will try implementing in a minute, hold on

Reply:
Is my latest commit what you meant?